lib/msal-common/src/app/module/AuthorizationCodeModule.ts

```diff
@@ -111,6 +111,36 @@ export class AuthorizationCodeModule extends AuthModule {
     }
 
     async acquireToken(request: TokenExchangeParameters): Promise<TokenResponse> {
+        const acquireTokenAuthority = (request && request.authority) ? AuthorityFactory.createInstance(request.authority, this.networkClient) : this.defaultAuthorityInstance;
+
+        let tokenEndpoint: string;
+        try {
+            tokenEndpoint = acquireTokenAuthority.tokenEndpoint;
+        } catch (e) {
+            const authErr: AuthError = e;
+            if (authErr.errorCode === ClientAuthErrorMessage.endpointResolutionError.code) {
+                await acquireTokenAuthority.resolveEndpointsAsync();
```

Comment:
If this is not sucessfull what happens?

Reply:
Same as what would happen if any resolve Endpoints fails, we receive an error response. We should probably handle this ourselves, but I'm not completely clear on all the error cases. I think we should revisit this when we talk through design for optimization of well-known endpoint.

Reply:
we should resolve the endpoint discovery as a separate issue, this is still an open question in the current library.